main/acle.md

```diff
@@ -4955,6 +5111,2370 @@ assertions, which may speed up access to aligned data (and will fault
+size. For example, if the code generator assumes that the vector register
+size is exactly 256 bits, it might use:
+
+``` asm
+  ADD X1, X2, #32
+```
+
+instead of:
+
+``` asm
+  ADDVL X1, X2, #1
+```
+
+Let SourceVL be the set of vector register sizes supported by a piece of
+SVE ACLE source code and ObjectVL be the set of vector register sizes
+for which the object code correctly implements the source code. In both
+cases, the set might be the set of all possible register sizes (referred
+to as “unrestricted” below).
+
+In general, object code created from SVE ACLE source code only works
+correctly when the runtime register size belongs to the intersection of
+SourceVL and ObjectVL. The behavior in other cases is undefined.
+
+The compiler might impose a fixed ObjectVL set for all code or provide a
+level of user control. It might also allow ObjectVL to vary within the
+program, either within the same translation unit or between different
+translation units. Whether and how this happens is currently outside the
+scope of ACLE.
+
+Note that the combination of a singleton SourceVL and an unrestricted
+ObjectVL makes sense: ACLE source code written for a single register
+size can still be compiled correctly without knowing what that size is,
+just like an assembler can assemble SVE code without knowing what
+register sizes the assembly code supports. Similarly, the combination of
+a singleton ObjectVL and an unrestricted SourceVL makes sense: ACLE
+source code written for all register sizes can still be compiled for
+only a single register size, perhaps to try to improve the performance
+of surrounding code. Other combinations are also possible.
+
+This section describes ACLE features related to singleton ObjectVLs.
+They are intended to help cases in which SourceVL is the same singleton.
+
+All the features are optional: a conforming ACLE implementation does not
+need to provide them.
+
+Note that the types defined in [SVE vector types](#sve-vector-types) and
+[SVE predicate types](#sve-predicate-types) are always sizeless types and
+are subject to the same restrictions for all SourceVL and for all ObjectVL.
+
+#### The `__ARM_FEATURE_SVE_BITS` macro
+
+As an optional extension, an ACLE implementation may set the
+preprocessor macro:
+
+``` c
+  __ARM_FEATURE_SVE_BITS
+```
+
+to a nonzero value N provided that:
+
+*   ObjectVL (see [Introduction to fixed-length SVE
+    types](#introduction-to-fixed-length-sve-types)
+    is a singleton vector register size of N bits; and
+
+*   it is possible to apply the `arm_sve_vector_bits(N)` attribute to
+    SVE vector types and SVE predicate types, as described in
+    [The `arm_sve_vector_bits` attribute](#the-arm_sve_vector_bits-attribute).
+
+If the implementation allows ObjectVL to be changed within a translation
+unit, it must ensure that the value of the macro remains accurate after
+each change. It is therefore possible for `__ARM_FEATURE_SVE_BITS` to
```

Comment:
```suggestion
each change. Therefore, it is possible for `__ARM_FEATURE_SVE_BITS` to
```

Reply:
I reworded this to avoid the “therefore” instead.